src/batch/src/executor/iceberg_scan.rs

```diff
@@ -199,7 +220,6 @@ impl BoxedExecutorBuilder for IcebergScanExecutorBuilder {
             .iter()
             .map(|split| SplitImpl::restore_from_bytes(split).unwrap())
             .collect_vec();
-        assert_eq!(split_list.len(), 1);
```

Comment:
I guess, the split_list is always with len = 1. That's why we don't need to introduce `IcebergScanNoDataExecutor`.

Reply:
Fixed.